src/bundle/Editor.ts

```diff
@@ -1,31 +1,32 @@
 import type {ReactNode} from 'react';
 
-import type {Extension as CodemirrorExtension} from '@codemirror/state';
 import {EditorView as CMEditorView} from '@codemirror/view';
 import {TextSelection} from 'prosemirror-state';
 import {EditorView as PMEditorView} from 'prosemirror-view';
 
-import {CommonEditor, MarkupString} from '../common';
-import {ActionStorage, WysiwygEditor, WysiwygEditorOptions} from '../core';
-import {ReactRenderStorage, RenderStorage} from '../extensions';
+import type {CommonEditor, MarkupString} from '../common';
+import {
+    type ActionStorage,
+    type EscapeConfig,
+    WysiwygEditor,
+    type WysiwygEditorOptions,
+} from '../core';
+import {ReactRenderStorage, type RenderStorage} from '../extensions';
 import {i18n} from '../i18n/bundle';
 import {logger} from '../logger';
-import {type CreateCodemirrorParams, createCodemirror} from '../markup/codemirror';
-import type {YfmLangOptions} from '../markup/codemirror/yfm';
-import {CodeEditor, Editor as MarkupEditor} from '../markup/editor';
-import {Emitter, Receiver, SafeEventEmitter} from '../utils/event-emitter';
+import {createCodemirror} from '../markup/codemirror';
+import {type CodeEditor, Editor as MarkupEditor} from '../markup/editor';
+import {type Emitter, type Receiver, SafeEventEmitter} from '../utils/event-emitter';
 import type {FileUploadHandler} from '../utils/upload';
 
-export type EditorMode = 'wysiwyg' | 'markup';
-export type SplitMode = false | 'horizontal' | 'vertical';
-export type EditorPreset = 'zero' | 'commonmark' | 'default' | 'yfm' | 'full';
-export type RenderPreview = ({
-    getValue,
-    mode,
-}: {
-    getValue: () => MarkupString;
-    mode: 'preview' | 'split';
-}) => ReactNode;
+import type {
+    MarkdownEditorMode as EditorMode,
```

Comment:
Maybe use `MarkdownEditorMode` everywhere, or is there a name conflict somewhere?

Reply:
> Maybe use `MarkdownEditorMode` everywhere

Yeah